pkg/fetch/windows/csaf/types.go

```diff
@@ -0,0 +1,226 @@
+	Summary  string `json:"summary"`
+	URL      string `json:"url"`
+}
+
+type Version string // pattern: ^(0|[1-9][0-9]*)$|^((0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?)$
+
+type Document struct {
+	Acknowledgments   []Acknowledgment `json:"acknowledgments,omitempty"`
+	AggregateSeverity *struct {
+		Namespace string `json:"namespace,omitempty"`
+		Text      string `json:"text"`
+	} `json:"aggregate_severity,omitempty"`
+	Category     string `json:"category"` // csaf_base, csaf_security_incident_response, csaf_informational_advisory, csaf_security_advisory, csaf_vex
+	CSAFVersion  string `json:"csaf_version"`
+	Distribution *struct {
+		Text string `json:"text,omitempty"`
+		TLP  *struct {
+			Label string `json:"label"` // enum: AMBER, GREEN, RED, WHITE
+			URL   string `json:"url,omitempty"`
+		} `json:"tlp,omitempty"`
+	} `json:"distribution,omitempty"`
+	Lang      Lang   `json:"lang,omitempty"`
+	Notes     []Note `json:"notes,omitempty"`
+	Publisher struct {
+		Category         string `json:"category"` // enum: coordinator, discoverer, other, translator, user, vendor
+		ContactDetails   string `json:"contact_details,omitempty"`
+		IssuingAuthority string `json:"issuing_authority,omitempty"`
+		Name             string `json:"name"`
+		Namespace        string `json:"namespace"`
+	} `json:"publisher"`
+	References []Reference `json:"references,omitempty"`
+	SourceLang Lang        `json:"source_lang,omitempty"`
+	Title      string      `json:"title"`
+	Tracking   struct {
+		Aliases            []string `json:"aliases,omitempty"`
+		CurrentReleaseDate string   `json:"current_release_date"`
+		Generator          struct {
+			Date   string `json:"date,omitempty"`
+			Engine struct {
+				Name    string `json:"name"`
+				Version string `json:"version,omitempty"`
+			} `json:"engine"`
+		} `json:"generator,omitempty"`
+		ID                 string `json:"id"`
+		InitialReleaseDate string `json:"initial_release_date"`
+		RevisionHistory    []struct {
+			Date          string  `json:"date"`
+			LegacyVersion string  `json:"legacy_version,omitempty"`
+			Number        Version `json:"number"`
+			Summary       string  `json:"summary"`
+		} `json:"revision_history"`
+		Status  string  `json:"status"` // enum: draft, final, interim
+		Version Version `json:"version"`
+	} `json:"tracking"`
+}
+
+type ProductTree struct {
+	Branches         []Branch          `json:"branches,omitempty"`
+	FullProductNames []FullProductName `json:"full_product_names,omitempty"`
+	ProductGroups    []struct {
+		GroupID    ProductGroupID `json:"group_id"`
+		ProductIDs []ProductID    `json:"product_ids"`
+		Summary    string         `json:"summary,omitempty"`
+	} `json:"product_groups,omitempty"`
+	Relationships []RelationShip `json:"relationships,omitempty"`
+}
+
+type RelationShip struct {
```

Comment:
Better to uncapitalize "s" because "relationship" is a sigle word?

Reply:
fixed: https://github.com/MaineK00n/vuls-data-update/compare/862456ae7e9df3ee953a5b2d46b7051dd79dcc3f..9530b96d610663827b3bc6bd5c951d36ff0b069d